### PR TITLE
Update to latest SDK snapshot

### PR DESCRIPTION
**Pull request by anuraaga**
WIP for after the next release. gRPC library instrumentation went swell, still debugging an issue with auto instrumentation of all libraries, some tests fail, others give a `VerifyError`

### CODE REVIEW

**Comment by anuraaga** [comment]
I think this error is the failure sign. @trask do you know if we need to reconfigure bytebuddy to instrument Java 8 classes? I'm guessing this ForJava6CapableVm is not appropriate for our Java 8 bytecode, but wild guess.

```
java.lang.IllegalStateException: Error invoking java.lang.instrument.Instrumentation#retransformClasses
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy$Dispatcher$ForJava6CapableVm.retransformClasses(AgentBuilder.java:6877)
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy$Collector$ForRetransformation.doApply(AgentBuilder.java:7148)
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy$Collector.apply(AgentBuilder.java:6993)
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy.apply(AgentBuilder.java:4855)
	at net.bytebuddy.agent.builder.AgentBuilder$Default.doInstall(AgentBuilder.java:9463)
	at net.bytebuddy.agent.builder.AgentBuilder$Default.installOn(AgentBuilder.java:9384)
	at net.bytebuddy.agent.builder.AgentBuilder$Default$Delegator.installOn(AgentBuilder.java:10986)
	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:157)
	at io.opentelemetry.instrumentation.test.AgentTestRunner.setupBeforeTests(AgentTestRunner.java:172)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.spockframework.util.ReflectionUtil.invokeMethod(ReflectionUtil.java:200)
	at org.spockframework.runtime.model.MethodInfo.invoke(MethodInfo.java:113)
	at org.spockframework.runtime.BaseSpecRunner.invokeRaw(BaseSpecRunner.java:484)
	at org.spockframework.runtime.BaseSpecRunner.invoke(BaseSpecRunner.java:467)
	at org.spockframework.runtime.BaseSpecRunner.doRunSetupSpec(BaseSpecRunner.java:177)
	at org.spockframework.runtime.BaseSpecRunner$3.invoke(BaseSpecRunner.java:161)
	at org.spockframework.runtime.BaseSpecRunner.invokeRaw(BaseSpecRunner.java:484)
	at org.spockframework.runtime.BaseSpecRunner.invoke(BaseSpecRunner.java:467)
	at org.spockframework.runtime.BaseSpecRunner.runSetupSpec(BaseSpecRunner.java:154)
	at org.spockframework.runtime.BaseSpecRunner.doRunSetupSpec(BaseSpecRunner.java:174)
	at org.spockframework.runtime.BaseSpecRunner$3.invoke(BaseSpecRunner.java:161)
	at org.spockframework.runtime.BaseSpecRunner.invokeRaw(BaseSpecRunner.java:484)
	at org.spockframework.runtime.BaseSpecRunner.invoke(BaseSpecRunner.java:467)
	at org.spockframework.runtime.BaseSpecRunner.runSetupSpec(BaseSpecRunner.java:154)
	at org.spockframework.runtime.BaseSpecRunner.doRunSetupSpec(BaseSpecRunner.java:174)
	at org.spockframework.runtime.BaseSpecRunner$3.invoke(BaseSpecRunner.java:161)
	at org.spockframework.runtime.BaseSpecRunner.invokeRaw(BaseSpecRunner.java:484)
	at org.spockframework.runtime.BaseSpecRunner.invoke(BaseSpecRunner.java:467)
	at org.spockframework.runtime.BaseSpecRunner.runSetupSpec(BaseSpecRunner.java:154)
	at org.spockframework.runtime.BaseSpecRunner.doRunSetupSpec(BaseSpecRunner.java:174)
	at org.spockframework.runtime.BaseSpecRunner$3.invoke(BaseSpecRunner.java:161)
	at org.spockframework.runtime.BaseSpecRunner.invokeRaw(BaseSpecRunner.java:484)
	at org.spockframework.runtime.BaseSpecRunner.invoke(BaseSpecRunner.java:467)
	at org.spockframework.runtime.BaseSpecRunner.runSetupSpec(BaseSpecRunner.java:154)
	at org.spockframework.runtime.BaseSpecRunner.runSetupSpec(BaseSpecRunner.java:149)
	at org.spockframework.runtime.BaseSpecRunner.doRunSpec(BaseSpecRunner.java:94)
	at org.spockframework.runtime.BaseSpecRunner$1.invoke(BaseSpecRunner.java:81)
	at org.spockframework.runtime.BaseSpecRunner.invokeRaw(BaseSpecRunner.java:484)
	at org.spockframework.runtime.BaseSpecRunner.invoke(BaseSpecRunner.java:467)
	at org.spockframework.runtime.BaseSpecRunner.runSpec(BaseSpecRunner.java:73)
	at org.spockframework.runtime.BaseSpecRunner.run(BaseSpecRunner.java:64)
	at org.spockframework.runtime.Sputnik.run(Sputnik.java:63)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:43)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:82)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:73)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:87)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:53)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:66)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:87)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:66)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:133)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.VerifyError: null
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at java.instrument/sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:167)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy$Dispatcher$ForJava6CapableVm.retransformClasses(AgentBuilder.java:6869)
	... 95 common frames omitted
```

**Comment by trask** [comment]
`ForJava6CapableVm` is good, the alternative is `ForLegacyVm` (Java 5 😱)

Can you add a breakpoint on throwing `VerifyError` and see what class is invalid?

**Comment by anuraaga** [comment]
@trask Apparently can't break at the exact class since it's in native code, and the exception doesn't have it either :( One of these classes

```
class org.gradle.internal.remote.internal.hub.MessageHub$ConnectionReceive
class org.gradle.internal.remote.internal.hub.MessageHub$ConnectionDispatch
class org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable
class org.gradle.internal.concurrent.ManagedExecutorImpl$1
class org.gradle.internal.remote.internal.hub.MessageHub$Handler
class org.gradle.internal.classloader.CachingClassLoader
class org.gradle.internal.classloader.MultiParentClassLoader
class org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker
class io.opentelemetry.javaagent.tooling.CommonTaskExecutor$PeriodicTask
class io.opentelemetry.javaagent.tooling.CommonTaskExecutor$ShutdownCallback
class com.blogspot.mydailyjava.weaklockfree.WeakConcurrentMap
class io.opentelemetry.javaagent.tooling.HelperInjector$1
class org.eclipse.jetty.io.nio.SelectorManager$SelectSet$1
class org.apache.http.impl.client.DefaultHttpClient
class org.apache.http.impl.client.AbstractHttpClient
class org.eclipse.jetty.io.nio.SelectChannelEndPoint$1
class org.eclipse.jetty.util.thread.ShutdownThread
class org.eclipse.jetty.server.AbstractConnector$Acceptor
class org.eclipse.jetty.io.nio.SelectorManager$1
class org.eclipse.jetty.util.thread.QueuedThreadPool$3
class org.eclipse.jetty.util.thread.QueuedThreadPool
class io.opentelemetry.instrumentation.test.base.HttpClientTest$__spock_initializeSharedFields_closure3$_closure31$_closure39
class io.opentelemetry.instrumentation.test.base.HttpClientTest$__spock_initializeSharedFields_closure3$_closure31$_closure38
class io.opentelemetry.instrumentation.test.base.HttpClientTest$__spock_initializeSharedFields_closure3$_closure31$_closure37
class io.opentelemetry.instrumentation.test.base.HttpClientTest$__spock_initializeSharedFields_closure3$_closure31$_closure36
class io.opentelemetry.instrumentation.test.base.HttpClientTest$__spock_initializeSharedFields_closure3$_closure31$_closure35
class io.opentelemetry.instrumentation.test.base.HttpClientTest$__spock_initializeSharedFields_closure3$_closure31$_closure34
class io.opentelemetry.instrumentation.test.base.HttpClientTest$__spock_initializeSharedFields_closure3$_closure31$_closure33
class io.opentelemetry.instrumentation.test.base.HttpClientTest$__spock_initializeSharedFields_closure3$_closure31$_closure32
class io.opentelemetry.instrumentation.test.base.HttpClientTest$__spock_initializeSharedFields_closure3$_closure31
class io.opentelemetry.instrumentation.test.server.http.TestHttpServer$_closure1
class io.opentelemetry.instrumentation.test.base.HttpClientTest$__spock_initializeSharedFields_closure3
class org.codehaus.groovy.runtime.callsite.CallSiteClassLoader
class org.codehaus.groovy.runtime.callsite.GroovySunClassLoader
class groovy.lang.GroovyClassLoader
class org.codehaus.groovy.runtime.DefaultGroovyMethods$2
class org.codehaus.groovy.runtime.MethodClosure
class groovy.lang.Closure$1
class groovy.lang.TrampolineClosure
class org.codehaus.groovy.runtime.ComposedClosure
class org.codehaus.groovy.runtime.CurriedClosure
class groovy.lang.Closure$WritableClosure
class org.spockframework.runtime.RunContext$2
class io.opentelemetry.instrumentation.test.base.HttpClientTest$__spock_feature_1_12_closure23
class groovy.lang.Closure
class worker.org.gradle.internal.classloader.FilteringClassLoader$RetrieveSystemPackagesClassLoader
class worker.org.gradle.internal.classloader.FilteringClassLoader
class io.opentelemetry.javaagent.bootstrap.AgentClassLoader
class java.util.concurrent.Executors$RunnableAdapter
class java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask
class java.util.concurrent.FutureTask
class java.util.concurrent.ScheduledThreadPoolExecutor
class java.util.concurrent.Executors$DelegatedExecutorService
class io.opentelemetry.javaagent.bootstrap.AgentClassLoader$BootstrapClassLoaderProxy
class java.util.concurrent.CompletableFuture$UniHandle
class java.util.concurrent.CompletableFuture$UniCompletion
class java.util.concurrent.CompletableFuture$Completion
class java.util.concurrent.ForkJoinTask
class java.util.concurrent.ForkJoinPool
class java.util.concurrent.ThreadPoolExecutor$Worker
class java.util.concurrent.ThreadPoolExecutor
class java.util.concurrent.AbstractExecutorService
class java.lang.ClassLoader
```

**Comment by trask** [comment]
eeek! we can't call static interface method `Context.current()` inside of advice that gets inlined into pre- Java 8 classes (that's why the `VerifyError`, e.g. since it's invalid Java 7 bytecode). at first I freaked out, what are we going to do! then I realized we can just create our own static util methods to wrap these calls 😅

File: instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/SpanInstrumentation.java [hunk 1]
```diff
@@ -23,25 +22,25 @@
...
-    return named("application.io.opentelemetry.trace.DefaultSpan");
+    return named("application.io.opentelemetry.trace.Span");
   }
 
   @Override
   public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
     return Collections.singletonMap(
-        isMethod().and(isPublic()).and(isStatic()).and(named("create")),
-        DefaultSpanInstrumentation.class.getName() + "$CreateAdvice");
+        isMethod().and(isPublic()).and(isStatic()).and(named("wrap")),
```
Comment (anuraaga) [inline instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/SpanInstrumentation.java:34]: @trask Not being able to get this instrumentation of a static interface method to kick in

**Comment by anuraaga** [comment]
I think this is mostly done but not sure - spring boot smoke tests don't work for me locally right now so can't run the whole build and have been playing some CI tag, haven't gotten to debug that yet. I can't do any more commits today I think, so if anyone has cycles to fix potentially remaining issues to get this landed tomorrow that would be nice :)

File: instrumentation/rmi/src/main/java/io/opentelemetry/javaagent/instrumentation/rmi/context/server/ContextDispatcher.java [hunk 2]
```diff
@@ -51,7 +50,7 @@ public void dispatch(Remote obj, RemoteCall call) throws IOException {
       ContextPayload payload = ContextPayload.read(in);
       if (payload != null) {
         Context context = extract(payload, GETTER);
-        SpanContext spanContext = getSpan(context).getContext();
+        SpanContext spanContext = io.opentelemetry.trace.Span.fromContext(context).getSpanContext();
```
Comment (trask) [inline instrumentation/rmi/src/main/java/io/opentelemetry/javaagent/instrumentation/rmi/context/server/ContextDispatcher.java:53]:
```suggestion
        SpanContext spanContext = Span.fromContext(context).getSpanContext();
```

File: instrumentation/scala-concurrent/src/slickTest/scala/SlickUtils.scala [hunk 3]
```diff
@@ -12,7 +13,7 @@ import scala.concurrent.{Await, Future}
 
 class SlickUtils {
   val TRACER: Tracer =
-    OpenTelemetry.getTracerProvider.get("io.opentelemetry.auto")
+    Java8Bridge.getGlobalTracer("io.opentelemetry.auto")
```
Comment (trask) [inline instrumentation/scala-concurrent/src/slickTest/scala/SlickUtils.scala:16]:
```suggestion
    // Java8Bridge is needed in order to support Scala 2.11 which targets Java 6 bytecode
    Java8Bridge.getGlobalTracer("io.opentelemetry.auto")
```

File: instrumentation/scala-concurrent/src/test/scala/ScalaConcurrentTests.scala [hunk 4]
```diff
@@ -14,14 +15,15 @@ import scala.concurrent.{Await, Future, Promise}
 
 class ScalaConcurrentTests {
   val TRACER: Tracer =
-    OpenTelemetry.getTracerProvider.get("io.opentelemetry.auto")
+    Java8Bridge.getGlobalTracer("io.opentelemetry.auto")
```
Comment (trask) [inline instrumentation/scala-concurrent/src/test/scala/ScalaConcurrentTests.scala:18]:
```suggestion
    // Java8Bridge is needed in order to support Scala 2.11 which targets Java 6 bytecode
    Java8Bridge.getGlobalTracer("io.opentelemetry.auto")
```

File: javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/Java8Bridge.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+
+/**
+ * A helper for accessing methods that rely on Java 8 bytecode such as static interface methods. In
+ * instrumentation, we may need to call these methods in code that is inlined into an instrumented
+ * class, however many times the instrumented class has been compiled to a previous version of
+ * bytecode and we cannot inline Java 8 method calls.
+ */
+public final class Java8Bridge {
```
Comment (trask) [inline javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/Java8Bridge.java:20]: too verbose?

though I think verbosity is ok for something confusing like this

```suggestion
public final class Java8BytecodeBridge {
```

File: javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/Java8Bridge.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+
+/**
+ * A helper for accessing methods that rely on Java 8 bytecode such as static interface methods. In
+ * instrumentation, we may need to call these methods in code that is inlined into an instrumented
+ * class, however many times the instrumented class has been compiled to a previous version of
+ * bytecode and we cannot inline Java 8 method calls.
```
Comment (trask) [inline javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/Java8Bridge.java:18]:
```suggestion
 * bytecode and so we cannot inline calls to static interface methods, as those were not supported prior to Java 8
 * and will lead to a class verification error.
```

File: javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/Java8Bridge.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+package io.opentelemetry.javaagent.instrumentation.api;
+
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+
+/**
+ * A helper for accessing methods that rely on Java 8 bytecode such as static interface methods. In
```
Comment (trask) [inline javaagent-api/src/main/java/io/opentelemetry/javaagent/instrumentation/api/Java8Bridge.java:15]:
```suggestion
 * A helper for accessing methods that rely on new Java 8 bytecode features such as calling a static interface method. In
```

**Review by trask — APPROVED** [review]
wow! great API changes, and massive update

File: instrumentation/grpc-1.5/testing/src/main/groovy/io/opentelemetry/instrumentation/grpc/v1_5/AbstractGrpcTest.groovy [hunk 8]
```diff
@@ -312,7 +311,7 @@ abstract class AbstractGrpcTest extends InstrumentationSpecification {
           responseObserver.onError(new AssertionError((Object) "context not preserved"))
           return
         }
-        if (!TracingContextUtils.getSpan(Context.current()).getContext().isValid()) {
+        if (!io.opentelemetry.trace.Span.fromContext(io.opentelemetry.context.Context.current()).getSpanContext().isValid()) {
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply instrumentation/grpc-1.5/testing/src/main/groovy/io/opentelemetry/instrumentation/grpc/v1_5/AbstractGrpcTest.groovy:314]: Thanks for finding them - yeah Groovy refactoring is painful 😿

File: instrumentation/kotlin-coroutines/src/test/kotlin/KotlinCoroutineTests.kt [hunk 9]
```diff
@@ -26,7 +26,7 @@ import kotlinx.coroutines.withTimeout
 import kotlinx.coroutines.yield
 
 class KotlinCoroutineTests(private val dispatcher: CoroutineDispatcher) {
-  val tracer: Tracer = OpenTelemetry.getTracer("io.opentelemetry.auto")
+  val tracer: Tracer = Java8Bridge.getGlobalTracer("io.opentelemetry.auto")
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply instrumentation/kotlin-coroutines/src/test/kotlin/KotlinCoroutineTests.kt:29]: Yeah Kotlin by default is Java 6 https://kotlinlang.org/docs/reference/faq.html#which-versions-of-jvm-does-kotlin-target

File: instrumentation/netty/netty-3.8/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/ChannelTraceContext.java [hunk 10]
```diff
@@ -49,7 +49,7 @@ public void setClientParentContext(Context clientParentContext) {
     this.clientParentContext = clientParentContext;
   }
 
-  public Context getContext() {
+  public Context getSpanContext() {
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply instrumentation/netty/netty-3.8/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v3_8/ChannelTraceContext.java:52]: 😱

File: instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/SpanInstrumentation.java [hunk 11]
```diff
@@ -23,25 +22,33 @@
...
+    }
+
+    @Advice.OnMethodExit
+    public static void methodExit(
+        @Advice.Argument(0) SpanContext applicationSpanContext,
+        @Advice.Return(readOnly = false) Span applicationSpan) {
       applicationSpan =
           Bridging.toApplication(
-              DefaultSpan.create(Bridging.toAgent(applicationSpan.getContext())));
+              io.opentelemetry.trace.Span.wrap(Bridging.toAgent(applicationSpanContext)));
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/SpanInstrumentation.java:51]: This isn't groovy :) think this is correct, we always import "application" types

[the remark this answers is not in the document]
Comment (trask) [reply instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/SpanInstrumentation.java:51]: 👍

**Pull request merged**